<h4 align="center" style="height:1123px;font-family: 'Times New Roman'"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “OPERATOR PADA PYTHON”

    Disusun Oleh :

    Zaky Afiq Taha

    2611533029

    Dosen Pengampu:

    Wahyudi.DR.. S.T.M.T

    Asisten Praktikum:

    SHERLY SUKMADIRA

<br>

 <img src="image/logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI

    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</b></h4>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">

<h2 align="center">KATA PENGANTAR</h2>
Puji dan syukur penulis panjatkan ke hadirat Tuhan Yang Maha Esa atas segala rahmat dan karunia-Nya, sehingga laporan praktikum yang berjudul **“Operasi pada Python”** ini dapat diselesaikan dengan baik. Laporan ini dibuat sebagai salah satu bentuk pemenuhan tugas praktikum sekaligus sebagai sarana untuk meningkatkan pemahaman mengenai berbagai operasi yang digunakan dalam bahasa pemrograman Python.

Melalui praktikum ini, penulis mempelajari berbagai jenis operasi dalam Python, seperti operasi aritmatika, perbandingan, logika, penugasan, bitwise, serta beberapa operasi lainnya. Pembelajaran tersebut diharapkan dapat membantu penulis memahami fungsi dan cara penggunaan setiap operator serta mampu menerapkannya dalam pembuatan program sederhana.

Penulis menyadari bahwa laporan ini masih jauh dari sempurna dan masih terdapat kekurangan dalam penyusunannya. Oleh sebab itu, penulis mengharapkan kritik dan saran yang membangun sebagai bahan evaluasi untuk penyusunan laporan selanjutnya. Semoga laporan ini dapat bermanfaat dan menambah wawasan bagi penulis maupun pembaca dalam memahami dasar-dasar operasi pada bahasa pemrograman Python.

<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
Zaky Afiq Taha
<br>
2611533029
</div>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB I<br>PENDAHULUAN</h1>

## 1.1 Latar Belakang
Pemrograman menjadi salah satu dasar penting yang perlu dikuasai dalam bidang Informatika. Dalam proses pembuatan program, terdapat berbagai operasi yang digunakan untuk melakukan perhitungan, membandingkan data, mengatur kondisi, serta memberikan atau mengubah nilai pada variabel. Python sebagai salah satu bahasa pemrograman menyediakan berbagai operator yang dapat digunakan untuk menjalankan operasi tersebut dengan penulisan sintaks yang relatif sederhana dan mudah dipahami.

Pada Python, terdapat berbagai macam operator, seperti operator aritmatika, perbandingan, logika, penugasan, bitwise, dan beberapa operator lainnya. Masing-masing operator mempunyai fungsi serta cara penggunaan yang berbeda sesuai dengan kebutuhan dalam suatu program. Oleh karena itu, memahami fungsi dan penggunaan operator merupakan salah satu hal dasar yang perlu dikuasai sebelum mempelajari pembuatan program yang lebih kompleks.

Melalui praktikum operasi pada Python, mahasiswa dapat mempelajari secara langsung cara kerja berbagai operator serta penerapannya dalam program sederhana. Dengan demikian, praktikum ini diharapkan dapat meningkatkan pemahaman mahasiswa terhadap konsep dasar operasi dan membantu membangun kemampuan pemrograman menggunakan Python.
## 1.2 Tujuan
1. Mengetahui berbagai jenis operasi pada Python.
2. Memahami fungsi dari setiap operator pada Python.
3. Mampu menggunakan operator dalam pembuatan program sederhana.
4. Mengetahui hasil dari setiap operasi yang dilakukan dalam Python.

## 1.3 Manfaat
1. Menambah pemahaman mengenai dasar-dasar bahasa pemrograman Python.
2. Membantu mahasiswa memahami penggunaan operator.
3. Melatih kemampuan dalam membuat program sederhana menggunakan Python.
4. Menjadi dasar untuk mempelajari pemrograman yang lebih kompleks.
</div>

---
<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB II<br>PEMBAHASAN</h1>

## 2.1 Dasar Teori
Python memiliki berbagai jenis operator yang digunakan untuk melakukan proses tertentu terhadap data dan variabel. Operator merupakan simbol atau kata khusus yang digunakan untuk melakukan operasi, seperti perhitungan, perbandingan, pemberian nilai, maupun pengolahan kondisi. Penggunaan operator menjadi salah satu dasar penting dalam pemrograman karena hampir setiap program membutuhkan operasi terhadap data.





1. Operator Aritmatika
Operator aritmatika digunakan untuk melakukan operasi perhitungan matematika pada Python. Operator yang termasuk di dalamnya adalah penjumlahan (+), pengurangan (-), perkalian (*), pembagian (/), pembagian bulat (//), modulus atau sisa pembagian (%), dan perpangkatan (**). Operator aritmatika dapat digunakan pada berbagai tipe data numerik, seperti integer dan float. Penggunaannya banyak ditemukan dalam perhitungan nilai, transaksi, luas bangun, dan berbagai proses perhitungan lainnya.
2. Operator Perbandingan
Operator perbandingan digunakan untuk membandingkan dua nilai atau lebih. Hasil dari operasi perbandingan berupa nilai Boolean, yaitu True atau False. Operator yang digunakan meliputi sama dengan (==), tidak sama dengan (!=), lebih besar (>), lebih kecil (<), lebih besar atau sama dengan (>=), serta lebih kecil atau sama dengan (<=). Operator ini biasanya digunakan dalam percabangan dan pemeriksaan kondisi pada suatu program.
3. Operator Logika
Operator logika digunakan untuk menggabungkan atau memanipulasi beberapa kondisi. Python memiliki tiga operator logika utama, yaitu and, or, dan not. Operator and menghasilkan nilai benar apabila semua kondisi yang diperiksa bernilai benar, sedangkan or menghasilkan nilai benar apabila terdapat setidaknya satu kondisi yang benar. Sementara itu, not digunakan untuk membalik nilai Boolean dari suatu kondisi. Operator logika banyak digunakan dalam proses pengambilan keputusan pada program.
4. Operator Penugasan
Operator penugasan digunakan untuk memberikan nilai kepada variabel atau mengubah nilai yang telah tersimpan di dalam variabel. Operator yang paling dasar adalah =, sedangkan operator gabungan lainnya antara lain +=, -=, *=, /=, //=, %= dan **=. Operator tersebut dapat membuat penulisan program menjadi lebih sederhana, terutama ketika suatu variabel perlu diperbarui berdasarkan nilai sebelumnya.
5. Operator Bitwise
Operator bitwise digunakan untuk melakukan operasi pada tingkat bit dari suatu bilangan integer. Python menyediakan beberapa operator bitwise, yaitu AND (&), OR (|), XOR (^), NOT (~), left shift (<<), dan right shift (>>). Operasi ini bekerja dengan memproses representasi biner dari suatu bilangan. Operator bitwise biasanya digunakan dalam pengolahan data tingkat rendah, manipulasi bit, sistem komputer, dan beberapa kebutuhan pemrograman tertentu.
6. Operator Keanggotaan
Operator keanggotaan digunakan untuk memeriksa apakah suatu nilai terdapat di dalam sebuah objek seperti string, list, tuple, atau set. Python memiliki operator in dan not in. Operator in menghasilkan True apabila suatu nilai ditemukan di dalam objek, sedangkan not in menghasilkan True apabila nilai tersebut tidak ditemukan. Operator ini dapat digunakan untuk melakukan pemeriksaan data dengan cara yang sederhana.
7. Operator Identitas
Operator identitas digunakan untuk mengetahui apakah dua variabel mengacu pada objek yang sama di dalam memori. Python menyediakan operator is dan is not. Operator is menghasilkan True apabila kedua variabel mengacu pada objek yang sama, sedangkan is not menghasilkan True apabila keduanya tidak mengacu pada objek yang sama. Operator identitas berbeda dengan operator == yang digunakan untuk membandingkan nilai.
## 2.2 Kode Program

### 2.2.1 Program aritmatika_2611533029
</div>

In [ ]:
# Buat file dengan nama aritmatika_NIM.py
# Buat program untuk operator aritmatika dalam Python
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer

angka1_3029 = int(input("Input angka-1: "))
angka2_3029 = int(input("Input angka-2: "))

# Penjumlahan
hasil_3029 = angka1_3029 + angka2_3029
print("\nOperator Penjumlahan")
print("Hasil =",hasil_3029)

# Pengurangan
hasil_3029 = angka1_3029 - angka2_3029
print("\nOperator Pengurangan")
print("Hasil =",hasil_3029)

# Perkalian
hasil_3029 = angka1_3029 * angka2_3029
print("\nOperator Perkalian")
print("Hasil =",hasil_3029)

# Pembagian, pembagian bulat, dan sisa bagi
if angka2_3029 != 0:
    hasil_3029 = angka1_3029 / angka2_3029
    print("\nOperator Pembagian")
    print("Hasil =",hasil_3029)

    hasil_3029 = angka1_3029 // angka2_3029
    print("\nOperator Pembagian Bulat")
    print("Hasil =",hasil_3029)

    hasil_3029 = angka1_3029 % angka2_3029
    print("\nOperator Sisa Bagi")
    print("Hasil =",hasil_3029)
else:
    print("Angka kedua tidak boleh bernilai 0.")

# Pangkat
hasil_3029 = angka1_3029 ** angka2_3029
print("\nOperator Pangkat")
print("Hasil =",hasil_3029)


Operator Penjumlahan
Hasil = 8

Operator Pengurangan
Hasil = -2

Operator Perkalian
Hasil = 15

Operator Pembagian
Hasil = 0.6

Operator Pembagian Bulat
Hasil = 0

Operator Sisa Bagi
Hasil = 3

Operator Pangkat
Hasil = 243


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3029 = int(input("Input angka-1: "))<br>
angka2_3029 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima data, sedangkan int() mengubah data yang dimasukkan menjadi tipe integer.
2. Operator Penjumlahan<br>
hasil_3029 = angka1_3029 + angka2_3029<br>
Operator + digunakan untuk menjumlahkan angka1_3029 dengan angka2_3029, kemudian hasilnya disimpan ke variabel hasil_3029.
3. Operator Pengurangan<br>
hasil_3029 = angka1_3029 - angka2_3029<br>
Operator - digunakan untuk mengurangi nilai angka1_3029 dengan angka2_3029.
4. Operator Perkalian<br>
hasil_3029 = angka1_3029 * angka2_3029<br>
Operator * digunakan untuk mengalikan kedua angka yang telah dimasukkan.
5. Operator Pembagian<br>
hasil_3029 = angka1_3029 / angka2_3029<br>
Operator / digunakan untuk melakukan pembagian. Hasil pembagian biasanya berupa tipe data float.
6. Operator Pembagian Bulat<br>
hasil_3029 = angka1_3029 // angka2_3029<br>
Operator // digunakan untuk melakukan pembagian dan mengambil hasil bilangan bulatnya saja, tanpa bagian desimal.
7. Operator Sisa Bagi<br>
hasil_3029 = angka1_3029 % angka2_3029<br>
Operator % digunakan untuk mendapatkan sisa dari hasil pembagian antara angka1_3029 dan angka2_3029.
8. Pengecekan angka kedua<br>
if angka2_3029 != 0:<br>
Program memeriksa apakah angka kedua tidak bernilai 0. Hal ini dilakukan karena pembagian dengan angka 0 tidak diperbolehkan.<br>
Jika angka2_3029 bernilai 0, program menjalankan:<br>
print("Angka kedua tidak boleh bernilai 0.")
9. Operator Pangkat<br>
hasil_3029 = angka1_3029 ** angka2_3029<br>
Operator ** digunakan untuk menghitung pangkat, yaitu angka1_3029 dipangkatkan dengan angka2_3029.


### 2.2.2 Program assigment_2611533029
</div>

In [ ]:
# Buat file dengan nama assigment_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator assigment dalam Python

angka1_3029 = int(input("Input angka-1: "))
angka2_3029 = int(input("Input angka-2: "))

print("\nNilai awal angka1 =",angka1_3029)
print("Nilai angka2 =",angka2_3029)

# Assigment biasa
hasil_3029 = angka1_3029 
print("\nAssigment biasa (=)")
print("Hasil =",hasil_3029)

# Assigment penambahan
hasil_3029 = angka1_3029 
hasil_3029 += angka2_3029 
print("\nAssigment penambahan (+=)")
print("Hasil =",hasil_3029)

# Assigment pengurangan
hasil_3029 = angka1_3029 
hasil_3029 -= angka2_3029 
print("\nAssigment pengurangan (-=)")
print("Hasil =",hasil_3029)

# Assigment perkalian
hasil_3029 = angka1_3029 
hasil_3029 *= angka2_3029
print("\nAssigment perkalian (*=)")
print("Hasil =",hasil_3029)

# Assigment pembagian, pembagian bulat, dan sisa bagi
if angka2_3029 != 0:
    hasil_3029 = angka1_3029 
    hasil_3029 /= angka2_3029
    print("\nAssigment pembagian (/=)")
    print("Hasil =",hasil_3029)
    # Operator tambahan
    hasil_3029 = angka1_3029 
    hasil_3029 //= angka2_3029
    print("\nAssigment pembagian bulat (//=)")
    print("Hasil =",hasil_3029)
    hasil_3029 = angka1_3029 
    hasil_3029 %= angka2_3029
    print("\nOperator sisa bagi (%=)")
    print("Hasil =",hasil_3029)
else:
    print("\nPembagian tidak dapat dilakukan.")
    print("Angka kedua tidak boleh bernilai 0.")

# Operator tambahan: assigment perpangkatan
hasil_3029 = angka1_3029 
hasil_3029 **= angka2_3029
print("\nOperator perpangkatan (**=)")
print("Hasil =",hasil_3029)


Nilai awal angka1 = 5
Nilai angka2 = 3

Assigment biasa (=)
Hasil = 5

Assigment penambahan (+=)
Hasil = 8

Assigment pengurangan (-=)
Hasil = 2

Assigment perkalian (*=)
Hasil = 15

Assigment pembagian (/=)
Hasil = 1.6666666666666667

Assigment pembagian bulat (//=)
Hasil = 1

Operator sisa bagi (%=)
Hasil = 2

Operator perpangkatan (**=)
Hasil = 125


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3029 = int(input("Input angka-1: "))<br>
angka2_3029 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan tersebut menjadi tipe data integer.
2. Menampilkan nilai awal<br>
print("\nNilai awal angka1 =", angka1_3029)<br>
print("Nilai angka2 =", angka2_3029)<br>
Bagian ini digunakan untuk menampilkan nilai awal dari angka1_3029 dan angka2_3029.
3. Assignment biasa (=)<br>
hasil_3029 = angka1_3029<br>
Operator = digunakan untuk memberikan atau menetapkan nilai dari angka1_3029 ke variabel hasil_3029.
4. Assignment penambahan (+=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 += angka2_3029<br>
Operator += digunakan untuk menambahkan nilai angka2_3029 ke hasil_3029.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 + angka2_3029
5. Assignment pengurangan (-=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 -= angka2_3029<br>
Operator -= digunakan untuk mengurangi hasil_3029 dengan nilai angka2_3029.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 - angka2_3029
6. Assignment perkalian (*=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 *= angka2_3029<br>
Operator *= digunakan untuk mengalikan hasil_3029 dengan angka2_3029.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 * angka2_3029
7. Assignment pembagian (/=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 /= angka2_3029<br>
Operator /= digunakan untuk membagi nilai hasil_3029 dengan angka2_3029.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 / angka2_3029<br>
Hasil pembagian menggunakan / biasanya berupa float.
8. Assignment pembagian bulat (//=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 //= angka2_3029<br>
Operator //= digunakan untuk melakukan pembagian bulat.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 // angka2_3029
9. Assignment sisa bagi (%=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 %= angka2_3029<br>
Operator %= digunakan untuk mendapatkan sisa hasil pembagian.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 % angka2_3029
10. Pengecekan nilai 0<br>
if angka2_3029 != 0:<br>
Program memeriksa apakah angka2_3029 tidak sama dengan 0. Pemeriksaan ini diperlukan karena operasi pembagian, pembagian bulat, dan sisa bagi tidak dapat dilakukan dengan pembagi 0.<br>
Jika nilainya 0, program menampilkan:<br>
print("\nPembagian tidak dapat dilakukan.")<br>
print("Angka kedua tidak boleh bernilai 0.")
11. Assignment perpangkatan (**=)<br>
hasil_3029 = angka1_3029<br>
hasil_3029 **= angka2_3029<br>
Operator **= digunakan untuk melakukan perpangkatan.<br>
Bentuk sederhananya:<br>
hasil_3029 = hasil_3029 ** angka2_3029
### 2.2.3 Program perbandingan_2611533029
</div>

In [9]:
# Buat file dengan nama perbandingan_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator perbandingan dalam Python

angka1_3029 = int(input("Input angka-1: "))
angka2_3029 = int(input("Input angka-2: "))

# Lebih besar dari
hasil_3029 = angka1_3029 > angka2_3029
print("\nOperator lebih besar dari")
print("angka1 > angka2 =",hasil_3029)

# Lebih kecil dari
hasil_3029 = angka1_3029 < angka2_3029
print("\nOperator lebih kecil dari")
print("angka1 < angka2 =",hasil_3029)

# Lebih besar dari atau sama dengan
hasil_3029 = angka1_3029 >= angka2_3029
print("\nOperator lebih besar dari atau sama dengan")
print("angka1 >= angka2 =",hasil_3029)

# Lebih kecil dari atau sama dengan
hasil_3029 = angka1_3029 <= angka2_3029
print("\nOperator lebih kecil dari atau sama dengan")
print("angka1 <= angka2 =",hasil_3029)

# Sama dengan
hasil_3029 = angka1_3029 == angka2_3029
print("\nOperator sama dengan")
print("angka1 == angka2 =",hasil_3029)

# Tidak sama dengan
hasil_3029 = angka1_3029 != angka2_3029
print("\nOperator tidak sama dengan")
print("angka1 != angka2 =",hasil_3029)

# Tambahan: perbandingan berantai dalam Python
hasil_3029 = 0 < angka1_3029 < 100
print("\nPerbandingan berantai")
print("0 < angka1_3029 < 100 =",hasil_3029)

hasil_3029 = 0 < angka2_3029 < 100
print("0 < angka2_3029 < 100 =",hasil_3029)


Operator lebih besar dari
angka1 > angka2 = False

Operator lebih kecil dari
angka1 < angka2 = True

Operator lebih besar dari atau sama dengan
angka1 >= angka2 = False

Operator lebih kecil dari atau sama dengan
angka1 <= angka2 = True

Operator sama dengan
angka1 == angka2 = False

Operator tidak sama dengan
angka1 != angka2 = True

Perbandingan berantai
0 < angka1_3029 < 100 = True
0 < angka2_3029 < 100 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3029 = int(input("Input angka-1: "))<br>
angka2_3029 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
2. Operator lebih besar dari (>)<br>
hasil_3029 = angka1_3029 > angka2_3029<br>
Operator > digunakan untuk mengecek apakah angka1_3029 lebih besar daripada angka2_3029.<br>
Hasilnya berupa nilai Boolean, yaitu True atau False.
3. Operator lebih kecil dari (<)<br>
hasil_3029 = angka1_3029 < angka2_3029<br>
Operator < digunakan untuk mengecek apakah angka1_3029 lebih kecil daripada angka2_3029.
4. Operator lebih besar dari atau sama dengan (>=)<br>
hasil_3029 = angka1_3029 >= angka2_3029<br>
Operator >= mengecek apakah angka1_3029 lebih besar atau sama dengan angka2_3029.
5. Operator lebih kecil dari atau sama dengan (<=)<br>
hasil_3029 = angka1_3029 <= angka2_3029<br>
Operator <= mengecek apakah angka1_3029 lebih kecil atau sama dengan angka2_3029.
6. Operator sama dengan (==)<br>
hasil_3029 = angka1_3029 == angka2_3029<br>
Operator == digunakan untuk mengecek apakah kedua angka memiliki nilai yang sama.<br>
Perlu dibedakan antara = dan ==.<br>
= digunakan untuk memberikan nilai, sedangkan == digunakan untuk membandingkan nilai.
7. Operator tidak sama dengan (!=)<br>
hasil_3029 = angka1_3029 != angka2_3029<br>
Operator != digunakan untuk mengecek apakah angka1_3029 dan angka2_3029 memiliki nilai yang berbeda.
8. Perbandingan berantai untuk angka pertama<br>
hasil_3029 = 0 < angka1_3029 < 100<br>
Kode tersebut merupakan perbandingan berantai (chained comparison). Artinya, program mengecek apakah angka1_3029 berada di antara 0 dan 100, dengan syarat:<br>
0 < angka1_3029 dan angka1_3029 < 100<br>
Jika kedua kondisi benar, hasilnya True.
9. Perbandingan berantai untuk angka kedua<br>
hasil_3029 = 0 < angka2_3029 < 100<br>
Bagian ini memiliki fungsi yang sama, tetapi digunakan untuk mengecek apakah angka2_3029 berada di antara 0 dan 100.

### 2.2.4 Program logika_2611533029
</div>

In [ ]:
# Buat file dengan nama logika_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: a1_1234
# Program ini menngunakan fungsi input()
# Program operator logika dalam Python

# Memasukan nilai boolean
# Input tidak peka terhadap huruf besar dan kecil
a1_3029 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_3029 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"

print("\nA1 = ",a1_3029)
print("A2 = ",a2_3029)

# Konjungsi: bernilai True jika keduanya True
hasil_3029 = a1_3029 and a2_3029
print("\nKonjungsi (AND)")
print("A1 and A2 =",hasil_3029)

# Disjungsi: bernilai True jika salah satunya False
hasil_3029 = a1_3029 or a2_3029
print("\nDisjungsi (OR)")
print("A1 or A2 =",hasil_3029)

# Negasi A1: membalik nilai A1
hasil_3029 = not a1_3029
print("\nNegasi A1(NOT)")
print("not A1",hasil_3029)

# Negasi A2: membalik nilai A2
hasil_3029 = not a2_3029
print("\nNegasi A2(NOT)")
print("not A2",hasil_3029)

# XOR: bernilai True jika kedua nilai berbeda
hasil_3029 = a1_3029 != a2_3029
print("\nDisjungsi Ekslusif (XOR)")
print("A1 XOR A2",hasil_3029)


A1 =  False
A2 =  True

Konjungsi (AND)
A1 and A2 = False

Disjungsi (OR)
A1 or A2 = True

Negasi A1(NOT)
not A1 True

Negasi A2(NOT)
not A2 False

Disjungsi Ekslusif (XOR)
A1 XOR A2 True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Memasukkan nilai Boolean<br>
a1_3029 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"<br>
a2_3029 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"<br>
Program meminta pengguna memasukkan nilai true atau false.<br>
input() menerima masukan dari pengguna.<br>
.strip() menghilangkan spasi di awal dan akhir.<br>
.lower() mengubah huruf menjadi huruf kecil sehingga TRUE, True, dan true dapat diproses dengan cara yang sama.<br>
== "true" membandingkan masukan dengan teks "true" dan menghasilkan nilai Boolean True atau False.
2. Menampilkan nilai Boolean<br>
print("\nA1 = ", a1_3029)<br>
print("A2 = ", a2_3029)<br>
Bagian ini menampilkan nilai Boolean yang telah dimasukkan dan dikonversi oleh program.
3. Konjungsi (AND)
hasil_3029 = a1_3029 and a2_3029<br>
Operator and digunakan untuk menggabungkan dua kondisi. Hasilnya akan True hanya jika kedua nilai bernilai True.<br>
Contoh:<br>
A1 = True<br>
A2 = True<br>
Hasil = True<br>
Jika salah satu atau keduanya False, hasilnya False.
4. Disjungsi (OR)<br>
hasil_3029 = a1_3029 or a2_3029<br>
Operator or menghasilkan True jika minimal salah satu nilai bernilai True.<br>
Hasilnya hanya False jika kedua nilai bernilai False.
5. Negasi A1 (NOT)<br>
hasil_3029 = not a1_3029<br>
Operator not digunakan untuk membalik nilai Boolean dari a1_3029.<br>
Jika A1 = True, maka not A1 = False.<br>
Jika A1 = False, maka not A1 = True.
6. Negasi A2 (NOT)<br>
hasil_3029 = not a2_3029<br>
Bagian ini memiliki fungsi yang sama dengan sebelumnya, tetapi digunakan untuk membalik nilai Boolean a2_3029.
7. Disjungsi Eksklusif (XOR)<br>
hasil_3029 = a1_3029 != a2_3029<br>
Operator XOR digunakan untuk menghasilkan True jika kedua nilai berbeda.

### 2.2.5 Program lainnya_2611533029
</div>

In [ ]:
# Buat file dengan nama lainnya_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()
# Program operator keanggotaan dan identitas

print("======================================")
print("1. OPERATOR KEANGGOTAAN")
print("======================================")

# Input beberapa data yang dipisahkan dengan koma
input_data_3029 = input("Masukkan beberapa angka, pisahkan dengan koma: ")

# Mengubah input menjadi list integer
data_3029 = [int(angka.strip()) for angka in input_data_3029.split(',')]

nilai_dicari_3029 = int(input("Masukkan angka yang ingin dicari: "))

# Operator in
hasil_3029 = nilai_dicari_3029 in data_3029
print("\nOperator keanggotaan IN")
print(nilai_dicari_3029,"in",data_3029,"=",hasil_3029)

# Operator not in
hasil_3029 = nilai_dicari_3029 not in data_3029
print("\nOperator keanggotaan NOT IN")
print(nilai_dicari_3029,"not in",data_3029,"=",hasil_3029)

print("======================================")
print("2. OPERATOR IDENTIAS")
print("======================================")

# objek1 menggunakan list dari input perngguna
object1_3029 = data_3029

# objek2 merujuk pada objek yang sama dengan objek1
object2_3029 = object1_3029

# objek3 memiliki isi sama, tetapi merupakan objek baru
object3_3029 = data_3029.copy()

print("object1 =",object1_3029)
print("object2 =",object2_3029)
print("object3 =",object3_3029)

# Operator is
hasil_3029 = object1_3029 is object2_3029
print("\nOperator identitas IS")
print("objek1 is objek2 =",hasil_3029)

# Operator is not
hasil_3029 = object1_3029 is not object2_3029
print("\nOperator identitas IS NOT")
print("objek1 is not objek2 =",hasil_3029)

# Membandingkan identitas dan nilai
print("\nPerbandingan identitas dan nilai")
print("objek1 is objek3 =",object1_3029 is object3_3029)
print("objek1 == objek3 =",object1_3029 == object3_3029)

1. OPERATOR KEANGGOTAAN

Operator keanggotaan IN
7 in [1, 23, 5, 6, 7, 1, 2, 6] = True

Operator keanggotaan NOT IN
7 not in [1, 23, 5, 6, 7, 1, 2, 6] = False
2. OPERATOR IDENTIAS
object1 = [1, 23, 5, 6, 7, 1, 2, 6]
object2 = [1, 23, 5, 6, 7, 1, 2, 6]
object3 = [1, 23, 5, 6, 7, 1, 2, 6]

Operator identitas IS
objek1 is objek2 = True

Operator identitas IS NOT
objek1 is not objek2 = False

Perbandingan identitas dan nilai
objek1 is objek3 = False
objek1 == objek3 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul program<br>
print("======================================")<br>
print("1. OPERATOR KEANGGOTAAN")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian Operator Keanggotaan agar output program lebih terstruktur.
2. Menerima input beberapa angka
input_data_3029 = input("Masukkan beberapa angka, pisahkan dengan koma: ")<br>
Program meminta pengguna memasukkan beberapa angka yang dipisahkan dengan tanda koma. Input tersebut awalnya disimpan sebagai string.<br>
Contoh:<br>
10, 20, 30, 40
3. Mengubah input menjadi list integer<br>
data_3029 = [int(angka.strip()) for angka in input_data_3029.split(',')]<br>
Kode ini mengubah input string menjadi list yang berisi integer.<br>
.split(',') memisahkan data berdasarkan tanda koma.<br>
.strip() menghilangkan spasi yang tidak diperlukan.<br>
int() mengubah setiap data menjadi integer.<br>
[ ... for ... ] digunakan untuk membuat list.<br>
Contoh:<br>
Input: 10, 20, 30<br>
Hasil: [10, 20, 30]
4. Memasukkan nilai yang ingin dicari<br>
nilai_dicari_3029 = int(input("Masukkan angka yang ingin dicari: "))<br>
Program meminta pengguna memasukkan angka yang akan dicari di dalam data_3029. Input dikonversi menjadi tipe data integer.
5. Operator keanggotaan in<br>
hasil_3029 = nilai_dicari_3029 in data_3029<br>
Operator in digunakan untuk mengecek apakah suatu nilai terdapat di dalam sebuah list.<br>
Jika nilai ditemukan, hasilnya True. Jika tidak ditemukan, hasilnya False.<br>
Contoh:<br>
data = [10, 20, 30]<br>
20 in data → True<br>
50 in data → False
6. Operator keanggotaan not in<br>
hasil_3029 = nilai_dicari_3029 not in data_3029<br>
Operator not in digunakan untuk mengecek apakah suatu nilai tidak terdapat di dalam list.<br>
Jika nilai tidak ditemukan, hasilnya True. Jika ditemukan, hasilnya False.
7. Menampilkan judul operator identitas<br>
print("======================================")<br>
print("2. OPERATOR IDENTIAS")<br>
print("======================================")<br>
Bagian ini menampilkan judul untuk bagian Operator Identitas.
8. Membuat object1<br>
object1_3029 = data_3029<br>
Variabel object1_3029 menunjuk ke objek list yang sama dengan data_3029.
9. Membuat object2<br>
object2_3029 = object1_3029<br>
Variabel object2_3029 menunjuk ke objek yang sama dengan object1_3029.<br>
Artinya, object1_3029 dan object2_3029 memiliki identitas objek yang sama.
10. Membuat object3<br>
object3_3029 = data_3029.copy()<br>
copy() membuat objek list baru yang memiliki isi yang sama dengan data_3029.<br>
Jadi, nilai object3_3029 sama dengan object1_3029, tetapi objeknya berbeda.
11. Menampilkan isi objek<br>
print("object1 =",object1_3029)<br>
print("object2 =",object2_3029)<br>
print("object3 =",object3_3029)<br>
Digunakan untuk menampilkan isi dari ketiga objek.<br>
Karena ketiganya memiliki isi yang sama, hasil yang ditampilkan akan terlihat sama.
12. Operator identitas is<br>
hasil_3029 = object1_3029 is object2_3029<br>
Operator is digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang sama.<br>
Karena object2_3029 dibuat dari object1_3029, hasilnya adalah:<br>
True
13. Operator identitas is not<br>
hasil_3029 = object1_3029 is not object2_3029<br>
Operator is not digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang berbeda.<br>
Karena object1_3029 dan object2_3029 menunjuk objek yang sama, hasilnya adalah:<br>
False
14. Membandingkan identitas dan nilai<br>
print("objek1 is objek3 =",object1_3029 is object3_3029)<br>
print("objek1 == objek3 =",object1_3029 == object3_3029)<br>
Baris pertama menggunakan is untuk membandingkan identitas objek. Karena object3_3029 dibuat menggunakan .copy(), objeknya berbeda sehingga hasilnya:<br>
False<br>
Baris kedua menggunakan == untuk membandingkan nilai atau isi objek. Karena isi object1_3029 dan object3_3029 sama, hasilnya:<br>
True

### 2.2.6 Program bitwise_2611533029
</div>

In [ ]:
# Buat file dengan nama bitwise_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()

print("======================================")
print("3. OPERATOR BITWISE")
print("======================================")

angka1_3029 = int(input("Masukkan angka bitwise-1: "))
angka2_3029 = int(input("Masukkan angka bitwise-2: "))

print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =",angka1_3029,"| biner =",bin(angka1_3029))
print("angka1 =",angka2_3029,"| biner =",bin(angka2_3029))

# Bitwise AND
hasil_3029 = angka1_3029 & angka2_3029
print("\nBitwise AND (&)")
print(angka1_3029,"&",angka2_3029,hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

# Bitwise OR
hasil_3029 = angka1_3029 | angka2_3029
print("\nBitwise OR (|)")
print(angka1_3029,"|",angka2_3029,hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

# Bitwise XOR
hasil_3029 = angka1_3029 ^ angka2_3029
print("\nBitwise XOR (^)")
print(angka1_3029,"^",angka2_3029,hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

# Bitwise NOT
hasil_3029 = ~angka1_3029
print("\nBitwise NOT (~)")
print(angka1_3029,"~",angka2_3029,hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

# Bitwise geser kiri
jumlah_geser_3029 = int(input("\nMasukkan jumlah pergeseran bit: "))

hasil_3029 = angka1_3029 << jumlah_geser_3029
print("\nBitwise geser kiri (<<)")
print(angka1_3029,"<<",jumlah_geser_3029,"=",hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

# Bitwise geser kanan
hasil_3029 = angka1_3029 >> jumlah_geser_3029
print("\nBitwise geser kanan (>>)")
print(angka1_3029,">>",jumlah_geser_3029,"=",hasil_3029)
print("Biner hasil =",bin(hasil_3029))
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

3. OPERATOR BITWISE

Angka dalam bentuk desimal dan biner
angka1 = 5 | biner = 0b101
angka1 = 4 | biner = 0b100

Bitwise AND (&)
5 & 4 4
Biner hasil = 0b100
Biner hasil (8 bit) = 00000100

Bitwise OR (|)
5 | 4 5
Biner hasil = 0b101
Biner hasil (8 bit) = 00000101

Bitwise XOR (^)
5 ^ 4 1
Biner hasil = 0b1
Biner hasil (8 bit) = 00000001

Bitwise NOT (~)
5 ~ 4 -6
Biner hasil = -0b110
Biner hasil (8 bit) = -0000110

Bitwise geser kiri (<<)
5 << 1 = 10
Biner hasil = 0b1010
Biner hasil (8 bit) = 00001010

Bitwise geser kanan (>>)
5 >> 1 = 2
Biner hasil = 0b10
Biner hasil (8 bit) = 00000010


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul<br>
print("======================================")<br>
print("3. OPERATOR BITWISE")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian program agar output terlihat lebih terstruktur.
2. Memasukkan dua angka<br>
angka1_3029 = int(input("Masukkan angka bitwise-1: "))<br>
angka2_3029 = int(input("Masukkan angka bitwise-2: "))<br>
Program meminta pengguna memasukkan dua angka. input() menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
3. Menampilkan angka desimal dan biner<br>
print("\nAngka dalam bentuk desimal dan biner")<br>
print("angka1 =",angka1_3029,"| biner =",bin(angka1_3029))<br>
print("angka1 =",angka2_3029,"| biner =",bin(angka2_3029))<br>
Menampilkan nilai angka dalam bentuk desimal dan biner. Fungsi bin() digunakan untuk mengubah bilangan integer menjadi bentuk biner.
4. Operasi AND (&)<br>
hasil_3029 = angka1_3029 & angka2_3029<br>
Operator & membandingkan setiap bit dari kedua angka. Hasil bit akan 1 jika kedua bit bernilai 1.<br>
print(angka1_3029,"&",angka2_3029,hasil_3029)<br>
print("Biner hasil =",bin(hasil_3029))<br>
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))<br>
Menampilkan hasil dalam bentuk desimal, biner biasa, dan biner dengan panjang 8 bit. format(..., "08b") digunakan untuk menampilkan biner minimal 8 digit.
5. Operasi OR (|)<br>
hasil_3029 = angka1_3029 | angka2_3029<br>
Operator | menghasilkan bit 1 jika salah satu atau kedua bit bernilai 1.<br>
print(angka1_3029,"|",angka2_3029,hasil_3029)<br>
print("Biner hasil =",bin(hasil_3029))<br>
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))<br>
Menampilkan hasil operasi dalam bentuk desimal dan biner.
6. Operasi XOR (^)<br>
hasil_3029 = angka1_3029 ^ angka2_3029<br>
Operator ^ menghasilkan bit 1 jika kedua bit berbeda.<br>
Contoh:<br>
0 ^ 0 = 0<br>
0 ^ 1 = 1<br>
1 ^ 0 = 1<br>
1 ^ 1 = 0<br>
Bagian print() berikutnya menampilkan hasil dalam bentuk desimal dan biner.
7. Operasi NOT (~)<br>
hasil_3029 = ~angka1_3029<br>
Operator ~ digunakan untuk membalik bit dari angka1_3029. Bit 1 menjadi 0 dan bit 0 menjadi 1.<br>
print(angka1_3029,"~",angka2_3029,hasil_3029)<br>
Menampilkan hasil operasi NOT.<br>
Namun, pada baris tersebut angka2_3029 ikut ditampilkan meskipun operasi NOT hanya dilakukan terhadap angka1_3029. Jika ingin output lebih sesuai, sebaiknya:<br>
print("~", angka1_3029, "=", hasil_3029)
8. Memasukkan jumlah pergeseran bit<br>
jumlah_geser_3029 = int(input("\nMasukkan jumlah pergeseran bit: "))<br>
Program meminta jumlah posisi bit yang ingin digeser. Nilai tersebut dikonversi menjadi integer.
9. Pergeseran bit ke kiri (<<)<br>
hasil_3029 = angka1_3029 << jumlah_geser_3029<br>
Operator << menggeser bit angka1_3029 ke sebelah kiri sebanyak jumlah_geser_3029 posisi.<br>
Secara sederhana, pergeseran kiri satu posisi biasanya sama dengan mengalikan bilangan dengan 2.<br>
Contoh:<br>
5 = 0101<br>
5 << 1 = 1010 = 10
10. Menampilkan hasil geser kiri<br>
print("\nBitwise geser kiri (<<)")<br>
print(angka1_3029,"<<",jumlah_geser_3029,"=",hasil_3029)<br>
print("Biner hasil =",bin(hasil_3029))<br>
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))<br>
Menampilkan hasil pergeseran dalam bentuk desimal, biner, dan biner 8 bit.
11. Pergeseran bit ke kanan (>>)<br>
hasil_3029 = angka1_3029 >> jumlah_geser_3029<br>
Operator >> menggeser bit angka1_3029 ke sebelah kanan sebanyak jumlah posisi yang ditentukan.<br>
Secara sederhana, untuk bilangan positif, pergeseran kanan satu posisi biasanya sama dengan membagi bilangan dengan 2 dan mengambil bagian bulatnya.<br>
Contoh:<br>
10 = 1010<br>
10 >> 1 = 0101 = 5
12. Menampilkan hasil geser kanan<br>
print("\nBitwise geser kiri (>>)")<br>
print(angka1_3029,">>",jumlah_geser_3029,"=",hasil_3029)<br>
print("Biner hasil =",bin(hasil_3029))<br>
print("Biner hasil (8 bit) =",format(hasil_3029,"08b"))

---
## Tugas Praktikum

**Soal :**

<div align="center">
<img src="image/soal_tugas.png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas3_2611533029
</div>

In [ ]:
print("=== SISTEM TRANSAKSI TOKO ===")
nama_3029 = input("Masukkan Nama Pelanggan : ")
status_3029 = input("Masukkan Status Pelanggan (member/nonmember) : ")
total_3029 = int(input("Masukkan Total Belanja : "))
jumlah_3029 = int(input("Masukkan Jumlah Barang : "))
promo_3029 = input("Masukkan Kode Promo : ")

print("\n=== DATA TRANSAKSI ===")
print(f"Nama Pelanggan       : {nama_3029}")
print(f"Status Pelanggan     : {status_3029}")
print(f"Total Belanja        : Rp{total_3029}")
print(f"Jumlah Barang        : {jumlah_3029}")
print(f"Kode Promo           : {promo_3029}")

kode_promo_3029 = ["HEMAT10", "HEMAT20", "PENCARIKATING", "IFHEBAT", "UNAND5G"] #list kode promo

syarat_total_3029 = total_3029 >= 200000 # apakah memenuhi syarat belanja
syarat_jumlah_3029 = jumlah_3029 >= 3 # apakah memenuhi syarat barang
status_valid_3029 = status_3029 == "member" # apakah user member
promo_valid_3029 = promo_3029 in kode_promo_3029 # apakah kode promo ada dalam list

print("\n=== HASIL VALIDASI ===")
print(f"Belanja >= Rp200000        : {syarat_total_3029}")
print(f"Jumlah Barang >= 3         : {syarat_jumlah_3029}")
print(f"Status Member              : {status_valid_3029}")
print(f"Kode Promo Tersedia        : {promo_valid_3029}")
print(f"Mendapatkan Diskon         : {syarat_jumlah_3029 or syarat_total_3029}")
print(f"Mendapatkan Promo          : {promo_valid_3029}")

diskon_3029 = 0.10

print("\n=== HASIL PERHITUNGAN ===")
print(f"Diskon                      : Rp{diskon_3029 * total_3029}")
print(f"Total Pembayaran            : Rp{total_3029 - total_3029 * diskon_3029}")
print(f"Rata-rata Harga Barang      : Rp{(total_3029 - total_3029 * diskon_3029)/jumlah_3029}")

print("\n=== HAK AKSES PELANGGAN ===")
print(f"Kode Hak Akses              : ...")
print(f"Member Access               : {status_valid_3029}")
print(f"Promo Access                : {promo_valid_3029}")
print(f"Free Shipping Access        : ...")

kode_transaksi_3029 = int(status_valid_3029) << 0 | int(syarat_total_3029) << 1 | int(syarat_jumlah_3029) << 2 | int(promo_valid_3029) << 3
kode_referensi_3029 = int(status_valid_3029) << 0 | int(syarat_total_3029) << 1 | int(promo_valid_3029) << 3

print("\n=== OPERASI BITWISE ===")
print("=== Kode Status Transaksi ===")
print(f"{format(int(status_valid_3029) << 0,"04b")} | {format(int(syarat_total_3029) << 1,"04b")} | {format(int(syarat_jumlah_3029) << 2,"04b")} | {format(int(promo_valid_3029) << 3,"04b")}")
print(f"Kode Biner   : {format(kode_transaksi_3029,"04b")}")
print(f"Kode Desimal : {kode_transaksi_3029}")

print("\n=== Pemeriksaan Status ===")
print("Cek Member")
print(f"{format(kode_transaksi_3029,"04b")} & {format(int(status_valid_3029) << 0,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_3029) & int(status_valid_3029) << 0,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3029) & int(status_valid_3029) << 0}")

print("Cek Promo")
print(f"{format(kode_transaksi_3029,"04b")} & {format(int(promo_valid_3029) << 3,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_3029) & int(promo_valid_3029) << 3,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3029) & int(promo_valid_3029) << 3}")

print("\n=== Perbandingan Status ===")
print(f"Kode Transaksi : {format(kode_transaksi_3029,"04b")}")
print(f"Kode Referensi : {format(kode_referensi_3029,"04b")}")
print(f"{format(kode_transaksi_3029,"04b")} ^ {format(kode_referensi_3029,"04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_3029) ^ (kode_referensi_3029),"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3029) ^ (kode_referensi_3029)}")

print("\n=== Shift ===")
print(f"{format(kode_transaksi_3029,"04b")} << 1")
print(f"Hasil Biner   : {format((kode_transaksi_3029) << 1,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3029) << 1}") 
print("=== SELESAI ===")

=== SISTEM TRANSAKSI TOKO ===

=== DATA TRANSAKSI ===
Nama Pelanggan       : zaky
Status Pelanggan     : member
Total Belanja        : Rp400000
Jumlah Barang        : 5
Kode Promo           : PENCARIKATING

=== HASIL VALIDASI ===
Belanja >= Rp200000        : True
Jumlah Barang >= 3         : True
Status Member              : True
Kode Promo Tersedia        : True
Mendapatkan Diskon         : True
Mendapatkan Promo          : True

=== HASIL PERHITUNGAN ===
Diskon                      : Rp40000.0
Total Pembayaran            : Rp360000.0
Rata-rata Harga Barang      : Rp72000.0

=== HAK AKSES PELANGGAN ===
Kode Hak Akses              : ...
Member Access               : True
Promo Access                : True
Free Shipping Access        : ...

=== OPERASI BITWISE ===
=== Kode Status Transaksi ===
0001 | 0010 | 0100 | 1000
Kode Biner   : 1111
Kode Desimal : 15

=== Pemeriksaan Status ===
Cek Member
1111 & 0001
Hasil Biner   : 0001
Hasil Desimal : 1
Cek Promo
1111 & 1000
Hasil Biner   : 1000

<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

**Penjelasan :** 
1. Menampilkan judul program<br>
print("=== SISTEM TRANSAKSI TOKO ===")<br>
Menampilkan judul utama program.
2. Memasukkan data pelanggan<br>
nama_3029 = input("Masukkan Nama Pelanggan : ")<br>
status_3029 = input("Masukkan Status Pelanggan (member/nonmember) : ")<br>
total_3029 = int(input("Masukkan Total Belanja : "))<br>
jumlah_3029 = int(input("Masukkan Jumlah Barang : "))<br>
promo_3029 = input("Masukkan Kode Promo : ")<br>
Program meminta nama, status pelanggan, total belanja, jumlah barang, dan kode promo. int() digunakan untuk mengubah total belanja dan jumlah barang menjadi integer.
3. Menampilkan data transaksi<br>
print(f"Nama Pelanggan       : {nama_3029}")<br>
print(f"Status Pelanggan     : {status_3029}")<br>
print(f"Total Belanja        : Rp{total_3029}")<br>
print(f"Jumlah Barang        : {jumlah_3029}")<br>
print(f"Kode Promo           : {promo_3029}")<br>
Menampilkan kembali data yang telah dimasukkan. f-string digunakan agar nilai variabel dapat langsung dimasukkan ke dalam teks.
4. Menentukan kode promo yang tersedia<br>
kode_promo_3029 = ["HEMAT10", "HEMAT20", "PENCARIKATING", "IFHEBAT", "UNAND5G"]<br>
Membuat sebuah list yang berisi kode-kode promo yang tersedia.
5. Mengecek syarat total belanja<br>
syarat_total_3029 = total_3029 >= 200000<br>
Mengecek apakah total belanja minimal Rp200.000. Hasilnya berupa True atau False.
6. Mengecek jumlah barang<br>
syarat_jumlah_3029 = jumlah_3029 >= 3<br>
Mengecek apakah pelanggan membeli minimal 3 barang.
7. Mengecek status member<br>
status_valid_3029 = status_3029 == "member"<br>
Mengecek apakah status pelanggan sama dengan "member". Jika sama persis, hasilnya True.
8. Mengecek kode promo<br>
promo_valid_3029 = promo_3029 in kode_promo_3029<br>
Operator in digunakan untuk mengecek apakah kode promo yang dimasukkan terdapat di dalam list kode_promo_3029.
9. Menampilkan hasil validasi<br>
print(f"Belanja >= Rp200000        : {syarat_total_3029}")<br>
print(f"Jumlah Barang >= 3         : {syarat_jumlah_3029}")<br>
print(f"Status Member              : {status_valid_3029}")<br>
print(f"Kode Promo Tersedia        : {promo_valid_3029}")<br>
Menampilkan hasil dari setiap pemeriksaan dalam bentuk True atau False.
10. Menentukan mendapatkan diskon<br>
print(f"Mendapatkan Diskon         : {syarat_jumlah_3029 or syarat_total_3029}")<br>
Operator or digunakan. Pelanggan dianggap mendapatkan diskon jika jumlah barang minimal 3 atau total belanja minimal Rp200.000.
11. Menentukan mendapatkan promo<br>
print(f"Mendapatkan Promo          : {promo_valid_3029}")<br>
Menampilkan apakah kode promo yang dimasukkan tersedia.
12. Menentukan nilai diskon<br>
diskon_3029 = 0.10<br>
Variabel diskon_3029 diberi nilai 0.10, yang berarti 10%.
13. Menghitung jumlah diskon<br>
print(f"Diskon : Rp{diskon_3029 * total_3029}")<br>
Menghitung nilai diskon dengan mengalikan total belanja dengan 10%.<br>
Contoh jika belanja Rp200.000:<br>
0.10 × 200000 = Rp20.000
14. Menghitung total pembayaran<br>
print(f"Total Pembayaran : Rp{total_3029 - total_3029 * diskon_3029}")<br>
Mengurangi total belanja dengan jumlah diskon untuk mendapatkan total yang harus dibayar.
15. Menghitung rata-rata harga barang<br>
print(f"Rata-rata Harga Barang : Rp{(total_3029 - total_3029 * diskon_3029)/jumlah_3029}")<br>
Menghitung rata-rata harga setiap barang setelah diskon dengan membagi total pembayaran dengan jumlah barang.
16. Menampilkan hak akses pelanggan<br>
print(f"Member Access : {status_valid_3029}")<br>
print(f"Promo Access : {promo_valid_3029}")<br>
Menampilkan hak akses berdasarkan status member dan validitas kode promo.<br>
Jika pelanggan adalah member, Member Access bernilai True. Jika kode promo tersedia, Promo Access bernilai True.
17. Membuat kode transaksi<br>
kode_transaksi_3029 = int(status_valid_3029) << 0 | int(syarat_total_3029) << 1 | int(syarat_jumlah_3029) << 2 | int<br>(promo_valid_3029) << 3
Bagian ini mengubah nilai Boolean menjadi angka dan menyusunnya menggunakan operasi bitwise.<br>
True dikonversi menjadi 1, sedangkan False menjadi 0.<br>
Setiap kondisi ditempatkan pada posisi bit berbeda:<br>
Status member → bit 0<br>
Total belanja → bit 1<br>
Jumlah barang → bit 2<br>
Promo → bit 3<br>
Operator << digunakan untuk menggeser bit, sedangkan | digunakan untuk menggabungkan semua bit tersebut.
18. Membuat kode referensi<br>
kode_referensi_3029 = int(status_valid_3029) << 0 | int(syarat_total_3029) << 1 | int(promo_valid_3029) << 3<br>
Membuat kode referensi dengan menggunakan status member, total belanja, dan promo. Kondisi jumlah barang tidak dimasukkan ke dalam kode ini.
19. Menampilkan kode biner setiap kondisi<br>
print(f"{format(int(status_valid_3029) << 0,"04b")} | {format(int(syarat_total_3029) << 1,"04b")} | {format(int<br>(syarat_jumlah_3029) << 2,"04b")} | {format(int(promo_valid_3029) << 3,"04b")}")
Menampilkan masing-masing kondisi dalam bentuk biner 4 bit.<br>
Fungsi:<br>
format(nilai, "04b")<br>
digunakan untuk mengubah nilai menjadi biner dengan panjang minimal 4 bit.
20. Menampilkan kode transaksi<br>
print(f"Kode Biner   : {format(kode_transaksi_3029,"04b")}")<br>
print(f"Kode Desimal : {kode_transaksi_3029}")<br>
Menampilkan kode transaksi dalam bentuk biner dan desimal.
21. Pemeriksaan status member<br>
print(f"{format(kode_transaksi_3029,"04b")} & {format(int(status_valid_3029) << 0,"04b")}")<br>
Menampilkan operasi bitwise AND antara kode transaksi dan bit status member.<br>
print(f"Hasil Biner   : {format((kode_transaksi_3029) & int(status_valid_3029) << 0,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_3029) & int(status_valid_3029) << 0}")<br>
Menghasilkan nilai berdasarkan pemeriksaan bit status member.
22. Pemeriksaan promo<br>
print(f"{format(kode_transaksi_3029,"04b")} & {format(int(promo_valid_3029) << 3,"04b")}")<br>
Menampilkan operasi AND untuk memeriksa bit promo.<br>
print(f"Hasil Biner   : {format((kode_transaksi_3029) & int(promo_valid_3029) << 3,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_3029) & int(promo_valid_3029) << 3}")<br>
Menampilkan hasil pemeriksaan promo dalam bentuk biner dan desimal.
23. Menampilkan kode transaksi dan referensi<br>
print(f"Kode Transaksi : {format(kode_transaksi_3029,"04b")}")<br>
print(f"Kode Referensi : {format(kode_referensi_3029,"04b")}")<br>
Menampilkan kedua kode dalam bentuk biner 4 bit untuk dibandingkan.
24. Membandingkan kode dengan XOR<br>
print(f"{format(kode_transaksi_3029,"04b")} ^ {format(kode_referensi_3029,"04b")}")<br>
Operator ^ digunakan untuk membandingkan bit dari kode transaksi dan kode referensi. Bit yang berbeda akan menghasilkan 1.<br>
print(f"Hasil Biner   : {format((kode_transaksi_3029) ^ (kode_referensi_3029),"04b")}")<br>
Baris ini menghitung XOR antara kode_transaksi_3029 dan kode_referensi_3029
25. Pergeseran kode transaksi<br>
print(f"{format(kode_transaksi_3029,"04b")} << 1")<br>
Menampilkan operasi pergeseran bit kode transaksi sebanyak 1 posisi ke kiri.<br>
print(f"Hasil Biner   : {format((kode_transaksi_3029) << 1,"04b")}")<br>
print(f"Hasil Desimal : {(kode_transaksi_3029) << 1}")<br>
Menggeser seluruh bit kode transaksi satu posisi ke kiri dan menampilkan hasilnya dalam bentuk biner dan desimal.
26. Menampilkan tanda selesai<br>
print("=== SELESAI ===")<br>
Menampilkan pesan bahwa program telah selesai dijalankan.

</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB III<br>KESIMPULAN</h1>

Berdasarkan praktikum yang telah dilaksanakan, dapat diketahui bahwa Python menyediakan berbagai macam operator yang memiliki fungsi berbeda dalam menjalankan suatu program. Operator tersebut mencakup operator aritmatika, perbandingan, logika, penugasan, bitwise, keanggotaan, identitas, dan beberapa jenis operator lainnya. Setiap operator dapat digunakan sesuai dengan kebutuhan dan proses yang ingin dilakukan dalam program.

Pemahaman terhadap berbagai operator tersebut membantu mahasiswa dalam membuat program sederhana, mulai dari melakukan perhitungan dan membandingkan data hingga mengolah kondisi serta memeriksa suatu nilai. Oleh karena itu, penguasaan penggunaan operator menjadi salah satu dasar yang penting untuk mendukung proses pembelajaran dan pengembangan program Python ke tingkat yang lebih lanjut.
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">DAFTAR PUSTAKA</h1>

1. GeeksforGeeks. (2026). Python Operators. GeeksforGeeks. https://www.geeksforgeeks.org/python/python-operators/

2. W3Schools. (2026). Python Operators. W3Schools. https://www.w3schools.com/python/python_operators.asp

3. Python Software Foundation. (2026). The Python Tutorial. Python Software Foundation. https://docs.python.org/3/tutorial/

4. Python Software Foundation. (2026). Python 3 Documentation. Python Software Foundation. https://docs.python.org/3/
</div>